### PR TITLE
[CI] Add code coverage report upload steps

### PR DESCRIPTION
**Pull request by vettoreldaniele**
Add code coverage reports to pull request CI. When the CI runs configurations with code coverage (`+coverage`) enabled, a code coverage report is generated and uploaded to a public Google Cloud Storage [bucket](https://storage.cloud.google.com/amdvlk-llpc-github-ci-artifacts-public/) for inspection. The link to the code coverage report is then posted to the pull request by the CI bot.

The Google Cloud Storage bucket has been configured so that it's read-only by the public and write access is only granted to the CI bot.

This should be tested by force-pushing when https://github.com/GPUOpen-Drivers/llpc/pull/1599 is merged and the nightly images have been updated.

### CODE REVIEW

**Comment by amdvlk-admin** [comment]
Can one of the admins verify this patch?

File: .github/workflows/check-amdllpc-docker.yml
Comment (kuhar) [inline .github/workflows/check-amdllpc-docker.yml:59]: Why `777`? Are permissions the issue or the file owner? If the former, could the generation script or docker step change the permissions?  If the latter, could we `chown` instead?

Comment (vettoreldaniele) [reply .github/workflows/check-amdllpc-docker.yml]: The owner is wrong. I changed it to `chown`.

**Comment by vettoreldaniele** [comment]
V1: change permissions fix to use `chown` rather than `chmod`.

File: .github/workflows/check-amdllpc-docker.yml
Comment (kuhar) [inline .github/workflows/check-amdllpc-docker.yml:59]: nit: can we put the `chown` outside of docker (IE run it in the runner itself)? I think this would be cleaner as docker does not have to learn about the runner username.

**Comment by vettoreldaniele** [comment]
V2: move `chown` in the github action step instead of inside docker.

**Review by kuhar — APPROVED** [review]
LGTM, thanks for the fixes

**Review by amdrexu — APPROVED** [review]

**Comment by kuhar** [comment]
retest this please

**Review by kuhar — CHANGES_REQUESTED** [review]
Let's leave it open until the new year, since nobody would fix it if there's some unexpected breakage.

**Comment by amdvlk-admin** [comment]
### Test summary for commit eb5e4a5358c24fd9d2514c1d21fcc480aa19e3b2
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>PAL: dev 44abc48d199dd28369bf7415eee1751c4bf616d2</li>
<li>SPVGEN: dev 580ade71d630a304805e9468ef22ad263c37d137</li>
<li>XGL: dev 38aa59bcf134d1cb340ddd2459b34b6eda798abe</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev c6950e03a4c97beadc88a8bf5bea3dc5aa2649ee</li>
</ul></details>

<details><summary>CTS tests (Failed: 0/225376)</summary><ul><li>Built with version 1.2.5.2</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 32512/56344 (57.7%)</li>
<li>Failed: 0/56344 (0.0%)</li>
<li>Not Supported: 23832/56344 (42.3%)</li><li>Warnings: 0/56344 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 32352/56344 (57.4%)</li>
<li>Failed: 0/56344 (0.0%)</li>
<li>Not Supported: 23991/56344 (42.6%)</li><li>Warnings: 1/56344 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 32688/56344 (58.0%)</li>
<li>Failed: 0/56344 (0.0%)</li>
<li>Not Supported: 23656/56344 (42.0%)</li><li>Warnings: 0/56344 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx103</summary><ul>

<li>Passed: 35669/56344 (63.3%)</li>
<li>Failed: 0/56344 (0.0%)</li>
<li>Not Supported: 20675/56344 (36.7%)</li><li>Warnings: 0/56344 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by vettoreldaniele** [comment]
V3: add GCloud SDK setup, only upload reports when this is the official repo (other repos don't have access to the GCloud secret).

**Comment by vettoreldaniele** [comment]
It looks like somebody with access to the repo's secrets has to set the `GCR_USER` and `GCR_KEY` secrets as input to the `check-amdllpc-docker.yml` workflow.

**Comment by kuhar** [comment]
> It looks like somebody with access to the repo's secrets has to set the `GCR_USER` and `GCR_KEY` secrets as input to the `check-amdllpc-docker.yml` workflow.

@vettoreldaniele I think this should be picked up automatically, same as in https://github.com/GPUOpen-Drivers/llpc/blob/b780577b54bcccb85c191d022b63ea5ce9cc72a4/.github/workflows/build-amdvlk-docker.yml#L41. Perhaps it's some GitHub protection against opening PRs that extract secrets and will fix itself once submitted? If so, could we check if the secrets are defined and skip the upload step if not?

**Comment by kuhar** [comment]
> That could be it. I thought there was a need for each workflow to be allow-listed for the secrets but it looks like that's not the case. I will add a check to skip if the secrets are not defined.

Does it work when you define these on your fork? It used to be able to test my workflows like this in the past.

**Comment by vettoreldaniele** [comment]
V4: add checks to skip steps if Google Cloud secrets cannot be accessed.

**Comment by vettoreldaniele** [comment]
I just found that GitHub says "A workflow job cannot access environment secrets until approval is granted by required approvers." https://docs.github.com/en/actions/security-guides/encrypted-secrets

Can you try approving this PR?

**Comment by kuhar** [comment]
> Can you try approving this PR?

Done. I think it also needs someone from AMD.

**Review by amdrexu — APPROVED** [review]

**Comment by kuhar** [comment]
retest this please

**Comment by kuhar** [comment]
test this please

**Comment by amdvlk-admin** [comment]
### Test summary for commit c6ffde411863cc006bbe389df628ebf34439b485
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 39f8940199e60c44d4211cf8165dfd12876316fa</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>PAL: dev bcec463efe5260776d486a5e3da0c549bc0a75d2</li>
<li>SPVGEN: dev 580ade71d630a304805e9468ef22ad263c37d137</li>
<li>XGL: dev 04534321e450d220d404a240a047c7e1cd82f3fd</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev aaa34e134023e703e197a48d8069cf3a6ebea629</li>
</ul></details>

<details><summary>CTS tests (Failed: 0/225376)</summary><ul><li>Built with version 1.2.5.2</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 32541/56344 (57.8%)</li>
<li>Failed: 0/56344 (0.0%)</li>
<li>Not Supported: 23803/56344 (42.2%)</li><li>Warnings: 0/56344 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 32381/56344 (57.5%)</li>
<li>Failed: 0/56344 (0.0%)</li>
<li>Not Supported: 23962/56344 (42.5%)</li><li>Warnings: 1/56344 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 32717/56344 (58.1%)</li>
<li>Failed: 0/56344 (0.0%)</li>
<li>Not Supported: 23627/56344 (41.9%)</li><li>Warnings: 0/56344 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx103</summary><ul>

<li>Passed: 35698/56344 (63.4%)</li>
<li>Failed: 0/56344 (0.0%)</li>
<li>Not Supported: 20646/56344 (36.6%)</li><li>Warnings: 0/56344 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by kuhar** [comment]
@vettoreldaniele Seems like the workflow file has a syntax error: https://github.com/GPUOpen-Drivers/llpc/actions/runs/1663669598.

Can you take a look?

**Pull request merged**
